src/plugins/interactive_setup/server/plugin.ts

```diff
@@ -0,0 +1,155 @@
+    }
+
+    core.http.registerRoutes('', (prebootRouter) => {
+      let completeSetup: (result: { shouldReloadConfig: boolean }) => void;
+
+      prebootRouter.post(
+        {
+          path: '/api/preboot/setup',
+          validate: {
+            body: schema.oneOf([
+              schema.object({
+                hosts: schema.arrayOf(schema.string(), { minSize: 1 }),
+                apiKey: schema.string(),
+                caFingerprint: schema.string(),
+              }),
+              schema.object({
+                hosts: schema.arrayOf(schema.string(), { minSize: 1 }),
+                username: schema.string(),
+                password: schema.string(),
+                caFingerprint: schema.string(),
+              }),
+            ]),
+          },
+          options: { authRequired: false },
+        },
+        async (context, request, response) => {
+          if (!core.preboot.isSetupOnHold()) {
+            return response.badRequest();
+          }
+
+          const client = core.elasticsearch
+            .createClient('data', {
+              hosts: request.body.hosts.map((host) => `https://${host}`),
+              ssl: { verificationMode: 'none' },
+              // caFingerprint: request.body.caFingerprint,
+            })
+            .asScoped({
+              headers: {
+                authorization:
+                  'apiKey' in request.body
+                    ? `ApiKey ${btoa(request.body.apiKey)}`
+                    : `Basic ${btoa(`${request.body.username}:${request.body.password}`)}`,
+              },
+            });
+
+          try {
+            const { body } = await client.asCurrentUser.transport.request({
+              method: 'GET',
+              path: '/_security/enroll/kibana',
+            });
+
+            const configPath = isDev
+              ? this.#initializerContext.env.configs.find((path) => path.includes('dev'))
+              : this.#initializerContext.env.configs[0];
+
+            if (!configPath) {
+              return response.customError({ statusCode: 500, body: 'Cannot find config file.' });
+            }
+
+            await fs.appendFile(
+              configPath,
+              `
+
+elasticsearch.hosts: [${request.body.hosts.map((host) => `"https://${host}"`).join(', ')}]
+elasticsearch.username: "kibana_system"
+elasticsearch.password: "${body.password}"
+elasticsearch.ssl.verificationMode: "none"
+`
+            );
+
+            completeSetup({ shouldReloadConfig: true });
+
+            return response.noContent();
+          } catch (err) {
+            return response.customError({ statusCode: 500, body: getDetailedErrorMessage(err) });
+          }
+        }
+      );
+
+      core.preboot.holdSetupUntilResolved(
+        `\n\n${chalk.bold(
+          chalk.whiteBright(`${chalk.cyanBright('i')} Kibana has not been configured.`)
+        )}\n\nGo to ${chalk.cyanBright(
+          chalk.underline('http://localhost:5601')
+        )} to get started.\n`,
```

Comment:
It would be nice to ensure these instructions are always visible when starting Kibana and don't get lost in the log output.

Reply:
Yeah, not sure if we can do anything about this. But there shouldn't be many logs after this log message (even in verbose mode) until you start interacting with the setup mode, so maybe it's okay.